conftest.py

```diff
@@ -1,223 +1,254 @@
-def reset_sensor_to_factory(make_valid_request, get_sensor_info):
-    def _reset_sensor_to_factory():
-        log.info("Send reset firmware request to sensor")
-        sensor_response = make_valid_request(SensorMethod.RESET_TO_FACTORY)
-        if sensor_response != "resetting":
-            raise RuntimeError("Sensor didn't respond to factory reset properly")
-
-        sensor_info = wait(
-            get_sensor_info, lambda x: isinstance(x, dict), tries=15, timeout=1
-        )
-        if not sensor_info:
-            raise RuntimeError("Sensor didn't reset to factory property")
-
-        return sensor_info
-
-    return _reset_sensor_to_factory
-
-
-@pytest.fixture(scope="session")
-def update_firmware(make_valid_request):
-    def _update_firmware():
-        log.info("Send firmware update request to sensor")
-        return make_valid_request(SensorMethod.UPDATE_FIRMWARE)
-
-    return _update_firmware
-
-
-@pytest.fixture(scope="session")
-def reboot(make_valid_request):
-    def _reboot():
-        log.info("Send reboot request to sensor")
-        return make_valid_request(SensorMethod.REBOOT)
-
-    return _reboot
-
-
-@pytest.fixture(scope="session")
-def factory_sensor_settings(reset_sensor_to_factory):
-    log.info("Reset sensor to factory defaults")
-    yield reset_sensor_to_factory()
-
-
-@pytest.fixture(autouse=True)
-def ensure_sensor_factory_settings(
-    factory_sensor_settings, reset_sensor_to_factory, get_sensor_info
-):
-    current_sensor_settings = get_sensor_info()
-    log.info("Ensure sensor has factory settings before starting test")
-    if current_sensor_settings != factory_sensor_settings:
-        log.info("Detected non-factory settings, resetting sensor")
-        reset_sensor_to_factory()
+import pytest
+from enum import Enum
+from dataclasses import dataclass
+import requests
+from typing import Callable
+from time import sleep
+import logging
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int
+
+    def __post_init__(self):
+        if not isinstance(self.name, str):
+            raise TypeError("'name' should be a string")
+        if self.name == "":
+            raise ValueError("'name' should not be empty")
+        if not isinstance(self.hid, str):
+            raise TypeError("'hid' should be a string")
+        if self.hid == "":
+            raise ValueError("'hid' should not be empty")
+        if not isinstance(self.model, str):
+            raise TypeError("'model' should be a string")
+        if self.model == "":
+            raise ValueError("'model' should not be empty")
+        if not isinstance(self.firmware_version, int):
+            raise TypeError("'firmware_version' should be an integer")
+        if not isinstance(self.reading_interval, int):
+            raise TypeError("'reading_interval' should be a int")
+        if self.reading_interval == "":

```

Comment:
Testing for a non-empty string is suitable for string properties but here we've got an integer. We're interested in checking that this is a valid value from sensor's standpoint. That would be seeing that it's equal or more than 1.